Implement handling of missing files

raemer opened the pull request:
#### Pull Request (PR) description
This pull requests add a paramter (`ignore_missing`) that allows controlling the behavior, if a path is not found on the target system. Currently, the modules fails at this point, as setfacl/getfacl fails.
With the new parameter, this can be configured to fail with error (as now; default), issue a notice in the agent's output or silently ignore the error.

#### Use case example
In our use case, we want to allow a monitoring user to read git repos in order to determine if they are at a specific commit. To allow this, we add `u:monitoring:rx` to the repo, via a hiera hash evaluated in a module present for each node. On some systems, this path may be missing (and is not monitored). We would like to be able to ignore this, as the repo also will not be checked.
We do not implenent the ACL in the monitoring module, as there may be requirements to add further ACLs to the same path.

bastelfreak commented:
jey @raemer, can you please rebase against our latest master branch? we did some work on the co pipeline.

raemer commented:
> jey @raemer, can you please rebase against our latest master branch? we did some work on the co pipeline.

Hi, I rebased and fixed a few indentation/code unreachable warnings.

bastelfreak commented:
@raemer thanks for the work. I rebased it again and will merge it.
